Bursts/matlab/getBurstOriginXYN.m

```diff
@@ -3,36 +3,42 @@
 %the centroid location of neurons that spiked the most in the first 100 
 %timesteps for each burst (brightest pixel in the starting frame)
 %
-%   Syntax: getBurstOriginXY(h5file)
+%   Syntax: getBurstOriginXY(h5dir)
 %   
 %   Input:  
-%   h5file  - BrainGrid result filename (e.g. tR_1.0--fE_0.90_10000)
+%   h5dir  - BrainGrid result filename
+%   the entire path is required for example
+%   '/CSSDIV/research/biocomputing/data/tR_1.90--fE_0.90'
 %
 %   Output: 
 %   <allBurstOriginXY.csv> - burst origin (x,y) location for every burst
 %   <allBurstOriginN.csv> - burst origin neuron number for every burst
 
 % Author:   Jewel Y. Lee (jewel87@uw.edu)
-% Last updated: 4/18/2018
-function [X, Y, N] = getBurstOriginXYN(h5file, o_bin, id)
-burstfile = [h5file, '/Binned/burst_', num2str(id), '.csv'];
-frames = csvread(burstfile);    
-xloc = hdf5read([h5file '.h5'], 'xloc');        % x location
-yloc = hdf5read([h5file '.h5'], 'yloc');        % y location
-grid = sqrt(length(xloc));                    	% grid size 
+% Last updated: 02/06/2022
+% Last updated by: Vu T. Tieu (vttieu1995@gmail.com)
+function [X, Y, N] = getBurstOriginXYN(h5dir, id, xloc, yloc)
+binnedBurstFramesFilePath = [h5dir, '/Binned/burst_', num2str(id), '.csv'];
+binnedBurstFrames = csvread(binnedBurstFramesFilePath);         % flattened image vectors
+originBin = 10;                                 
+grid = sqrt(length(xloc));                    	% grid size
+
 % find brightest pixel (neuron that spikes the most in a time bin)
-largest = max(frames(:,o_bin)); 
-while (largest < 2)  
-    o_bin = o_bin + 1;
-    largest = max(frames(:,o_bin)); 
+brightestPixel = max(binnedBurstFrames(:,originBin));      % originBin starts at 10
+while (brightestPixel < 2)                                 
+    originBin = originBin + 1;
+    % finds largest element in each column starting at column 10
+    brightestPixel = max(binnedBurstFrames(:,originBin));  
```

Comment:
It would definitely be faster to call max once rather than thousands of time. I can add a note to move it and pass it in for future pull request. I've added some more documentation on the algorithm here.